api/middleware/authenticate.js

```diff
@@ -1,12 +1,19 @@
 const jwt = require('jsonwebtoken');
 const { jwtSecret } = require('../../config/secrets');
+const { getUser } = require('../../data/helpers');
 
-function authenticate(req, res, next) {
+async function authenticate(req, res, next) {
   try {
     const header = req.get('Authorization');
     if (header) {
       const token = header.split(' ')[1];
+      if (!token) throw new Error('Token must be set on Authorization header.');
```

Comment:
Excellent. We're gonna forget to put that in I promise you.

We also have a similar error in line 22. Is this an unintentional double-up? Or are there two cases you have to handle separately to throw the error?

Reply:
This is something I'll definitely be refactoring to make more DRY.  Essentially the one on line 22 is just a catch-all if, in some alternate universe, the code somehow executes to that point.